python/cog/types.py

```diff
@@ -195,22 +194,17 @@ def unlink(self, missing_ok: bool = False) -> None:
                     raise
 
 
-# we would prefer URLFile to stay lazy
-# except... that doesn't really work with httpx?
-
-
 class URLFile(io.IOBase):
     """
     URLFile is a proxy object for a :class:`urllib3.response.HTTPResponse`
     object that is created lazily. It's a file-like object constructed from a
     URL that can survive pickling/unpickling.
-
-    This is the only place Cog uses requests
     """
 
     __slots__ = ("__target__", "__url__")
 
     def __init__(self, url: str) -> None:
+        object.__setattr__(self, "name", os.path.basename(url))
```

Comment:
Ditto my comment on the other PR. This isn't quite enough work to extract a sensible name from a URL.